packages/trader/src/App/Components/Form/TimePicker/__tests__/dialog.spec.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Dialog from '../dialog';
+import moment from 'moment';
+
+describe('<Dialog />', () => {
+    const startTimes = [moment('2023-11-21T08:00:00')];
+    const endTimes = [moment('2023-11-22T12:00:00')];
+    const onChangeMock = jest.fn();
+    const className = 'testClassName';
+    const preClass = 'testPreClass';
+    const selectedTime = '23:45';
+
+    it('Should render minute and hours dialogs', () => {
+        render(
+            <Dialog
+                className={className}
+                preClass={preClass}
+                selected_time={selectedTime}
+                start_times={startTimes}
+                end_times={endTimes}
+                onChange={onChangeMock}
+            />
+        );
+        const hourElement = screen.getByText('23');
+        const minuteElement = screen.getByText('45');
+
+        expect(screen.getByText(/hour/i)).toBeInTheDocument();
+        expect(screen.getByText(/minute/i)).toBeInTheDocument();
+        expect(hourElement).toBeInTheDocument();
+        expect(minuteElement).toBeInTheDocument();
+        expect(minuteElement).toHaveClass('testPreClass__selector-list-item--selected');
+        expect(hourElement).toHaveClass('testPreClass__selector-list-item--selected');
+    });
+    it('Selecting disabled hour and minute does not call onChange function', () => {
+        render(
+            <Dialog
+                className={className}
+                preClass={preClass}
+                selected_time={selectedTime}
+                start_times={startTimes}
+                end_times={endTimes}
+                onChange={onChangeMock}
+            />
+        );
+        const disabledHourElement = screen.getByText('12');
+        const disabledMinuteElement = screen.getByText('45');
```

Comment:
Didn't catch the idea: in the previous test case you get same minutes and check if they have selected className; here you are checking if they can't be selected

Reply:
im checking if onchange gets called or not. Should not call on change if I click on 12 and 45